src/main/webapp/app/modeling-editor/modeling-editor.component.ts

```diff
@@ -117,4 +125,63 @@ export class ModelingEditorComponent implements AfterViewInit, OnDestroy, OnChan
             this.apollonEditor.destroy();
         }
     }
+
+    /**
+     * Resets the UML model for the guided tour by removing the elements, relationships and assessments
+     * @param umlModel the model that should be reset
+     */
+    private resetUMLModelForGuidedTour(umlModel: UMLModel): void {
+        if (umlModel) {
+            umlModel.elements = [];
+            umlModel.relationships = [];
+            umlModel.assessments = [];
+        }
+        this.initializeApollonEditor();
+    }
+
+    /**
+     * Subscribes to the guided tour service
+     */
+    private subscribeForUMLModelReset() {
+        this.guidedTourService.resetUMLModel().subscribe(reset => {
+            if (reset) {
+                this.resetUMLModelForGuidedTour(this.umlModel);
+            }
+        });
+    }
+
+    /**
+     * Assess the model for the modeling guided tutorial
+     * @param umlName  the identifier of the UML element that has to be assessed
+     * @param umlModel  the current UML model in the editor
+     */
+    assessModelForGuidedTour(umlName: string, umlModel: UMLModel): void {
+        // Find the required UML classes
+        const personClass = umlModel.elements.find(element => element.name.trim() === personUML.name && element.type === UMLElementType.Class);
```

Comment:
This line could be refactored into its own method (since you are using pretty much the same code five times here)